### PR TITLE
RUN: Fix synchronous execution on EDT in `computeSysroot`

### PR DESCRIPTION
**Pull request by ortem**

### CODE REVIEW

**Comment by vlad20012** [comment]
Where and why it is called from EDT?

**Comment by ortem** [comment]
@vlad20012 in `org.rust.debugger.runconfig.RsDebugRunner#getRunContentDescriptor`, so it affects only CLion.

File: src/main/kotlin/org/rust/cargo/runconfig/CargoRunStateBase.kt [hunk 1]
```diff
@@ -26,13 +26,15 @@ abstract class CargoRunStateBase(
...
-    private val toolchain: RustToolchain = config.toolchain
+    val toolchain: RustToolchain = config.toolchain
     val commandLine: CargoCommandLine = config.cmd
     val cargoProject: CargoProject? = CargoCommandConfiguration.findCargoProject(
         environment.project,
         commandLine.additionalArguments,
         commandLine.workingDirectory
     )
+    private val workingDirectory = cargoProject?.workingDirectory
+    val computeSysroot: () -> String? = { workingDirectory?.let { toolchain.getSysroot(it) } }
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/runconfig/CargoRunStateBase.kt]: Why did you convert `computeSysroot` into property?
Let's revert this change and use function reference syntax instead

File: debugger/src/main/kotlin/org/rust/debugger/runconfig/RsLocalDebugProcess.kt [hunk 2]
```diff
@@ -82,9 +102,14 @@ class RsLocalDebugProcess(
...
 
-    private fun GDBDriver.loadPrettyPrinters(threadId: Long, frameIndex: Int, sysroot: String?, renderers: GDBRenderers) {
+    private fun GDBDriver.loadPrettyPrinters(threadId: Long, frameIndex: Int, renderers: GDBRenderers) {
         when (renderers) {
             GDBRenderers.COMPILER -> {
+                val sysroot = sysroot
+                if (sysroot == null) {
+                    project.showBalloon("Cannot load rustc renderers", NotificationType.WARNING)
+                    return
+                }
```
Comment (Undin) [inline debugger/src/main/kotlin/org/rust/debugger/runconfig/RsLocalDebugProcess.kt]: You have three similar code block
Let's extract it into separate function

File: debugger/src/main/kotlin/org/rust/debugger/runconfig/RsLocalDebugProcess.kt [hunk 3]
```diff
@@ -24,17 +26,34 @@ class RsLocalDebugProcess(
     consoleBuilder: TextConsoleBuilder
 ) : CidrLocalDebugProcess(parameters, debugSession, consoleBuilder) {
 
-    fun loadPrettyPrinters(sysroot: String?, lldbRenderers: LLDBRenderers, gdbRenderers: GDBRenderers) {
+    @Volatile
+    private var sysroot: String? = null
```
Comment (mchernyavsky) [inline debugger/src/main/kotlin/org/rust/debugger/runconfig/RsLocalDebugProcess.kt]:
```kotlin
private val sysroot: String? by lazy(computeSysroot)
```

File: src/main/kotlin/org/rust/cargo/runconfig/CargoRunStateBase.kt [hunk 4]
```diff
@@ -26,13 +26,14 @@ abstract class CargoRunStateBase(
...
 ) : CommandLineState(environment) {
-    private val toolchain: RustToolchain = config.toolchain
+    val toolchain: RustToolchain = config.toolchain
     val commandLine: CargoCommandLine = config.cmd
     val cargoProject: CargoProject? = CargoCommandConfiguration.findCargoProject(
         environment.project,
         commandLine.additionalArguments,
         commandLine.workingDirectory
     )
+    private val workingDirectory = cargoProject?.workingDirectory
```
Comment (mchernyavsky) [inline src/main/kotlin/org/rust/cargo/runconfig/CargoRunStateBase.kt]:
```kotlin
private val workingDirectory: String? get() = cargoProject?.workingDirectory
```
